scripts/generate-releases.sh

```diff
@@ -2,64 +2,84 @@
 
 set -euo pipefail
 
-if [[ "$#" -lt 1 || "$#" -gt 3 ]]; then
-    echo "USAGE: ./generate-releases.sh <ENVIRONMENT> [COMMIT] [BRANCH]" >&2
-    echo "" >&2
-    echo "ENVIRONMENT - allowed values: staging|prod" >&2
-    echo "COMMIT - a 40 character-long SHA of the commit to pull Snapshots only with this commit label for the Release. Default: currently checked out commit" >&2
-    echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
-    echo "" >&2
-    echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
-    exit 1
-fi
-
-ENVIRONMENT="$1"
-COMMIT="${2:-$(git rev-parse HEAD)}"
-BRANCH="${3:-$(git rev-parse --abbrev-ref HEAD)}"
-
-release_name="${ENVIRONMENT}-$(git rev-parse --short HEAD)-$(date +'%Y-%m-%d-%H-%M')"
-# Fetch the list of snapshots for COMMIT and BRANCH. 
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+source "$SCRIPT_DIR/helpers.sh"
+
+# Describe the usage of the script if not enough arguments are provided.
+usage() {
+    if [[ "$#" -lt 1 || "$#" -gt 3 ]]; then
+        echo "USAGE: ./generate-releases.sh <ENVIRONMENT> [COMMIT] [BRANCH]" >&2
+        echo "" >&2
+        echo "ENVIRONMENT - allowed values: staging|prod" >&2
+        echo "COMMIT - a SHA of the commit to pull Snapshots only with this commit label for the Release." >&2
+        echo "If provided commit SHA is less than 40 characters then it will be expanded to a full 40-characters long SHA. Default: currently checked out commit" >&2
+        echo "BRANCH - an optional parameter to specify git branch name for filtering snapshots by having branch name in annotations. Default: currently checked out branch" >&2
+        echo "" >&2
+        echo "You must have your KUBECONFIG point to the Konflux cluster, see https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#HowtoeverythingKonfluxforRHACS-GettingocCLItoworkwithKonflux." >&2
+        return 1
+    fi
+}
+
+# Validate environment input and ensure it is either 'staging' or 'prod' (for master branch only).
+validate_environment() {
+    environment="$1"
+    branch="$2"
```

Comment:
These can be `local`-s to not overwrite the global ones:
```suggestion
    local -r environment="$1"
    local -r branch="$2"
```
`-r` makes them read-only.